### PR TITLE
[FrameworkBundle] Add support to 307/308 HTTP status codes in RedirectController

### PR DESCRIPTION
**Pull request by ZipoKing**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |  #26171
| License       | MIT
| Doc PR        | 

With this PR `RedirectController` will allow to create redirections with use of 307/308 HTTP status codes together with 301/302. Related RFC documents:
* https://tools.ietf.org/html/rfc7231
* https://tools.ietf.org/html/rfc7538

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php [hunk 1]
```diff
@@ -46,29 +46,36 @@ public function __construct(UrlGeneratorInterface $router = null, int $httpPort
...
+     * @param Request    $request           The request instance
+     * @param string     $route             The route name to redirect to
+     * @param bool       $permanent         Whether the redirection is permanent
+     * @param bool|array $ignoreAttributes  Whether to ignore attributes or an array of attributes to ignore
+     * @param bool       $keepRequestMethod Wheter redirect action should keep HTTP request method
      *
      * @throws HttpException In case the route name is empty
      */
-    public function redirectAction(Request $request, string $route, bool $permanent = false, $ignoreAttributes = false): Response
+    public function redirectAction(Request $request, string $route, bool $permanent = false, $ignoreAttributes = false, bool $keepRequestMethod = false): Response
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php]: for other reviewers: the class is final, so this is not a BC break

File: src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php [hunk 2]
```diff
@@ -46,29 +46,36 @@ public function __construct(UrlGeneratorInterface $router = null, int $httpPort
...
     {
         if ('' == $route) {
-            throw new HttpException($permanent ? 410 : 404);
+            throw new HttpException($permanent ? Response::HTTP_GONE : Response::HTTP_NOT_FOUND);
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php]: we prefer explicit numbers; please revert, and use numbers everywhere in the PR

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
thanks for working on this; see comments

File: src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php (hunk 2 above)
Comment (ZipoKing) [reply src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php]: @nicolas-grekas Change reverted

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ZipoKing.

**Comment by ZipoKing** [comment]
@stof will prepare some docs PR later today

**Pull request closed without merging**